kornia/enhance/adjust.py

```diff
@@ -250,6 +250,7 @@ def _solarize(input: torch.Tensor, thresholds: Union[float, torch.Tensor] = 0.5)
             f"threshholds must be a 1-d vector of shape ({input.size(0)},). Got {thresholds}"
         # TODO: I am not happy about this line, but no easy to do batch-wise operation
         thresholds = torch.stack([x.expand(*input.shape[1:]) for x in thresholds])
+        thresholds = thresholds.to(input.device).to(input.dtype)
```

Comment:
```suggestion
        thresholds = thresholds.to(input.device).to(input.dtype)
        thresholds = torch.stack([x.expand(*input.shape[1:]) for x in thresholds])
```
It is better to make the calculation happens inside your pointed device as well.

Reply:
Also `thresholds = thresholds.to(input)` will work